cmd/agent/app/agent.go

```diff
@@ -405,8 +405,8 @@ func generateClusterInControllerPlane(opts util.ClusterRegisterOption) (*cluster
 			cluster.Spec.Provider = opts.ClusterProvider
 		}
 
-		if opts.ClusterZone != "" {
-			cluster.Spec.Zone = opts.ClusterZone
+		if len(opts.ClusterZones) > 0 {
+			cluster.Spec.Zones = opts.ClusterZones
```

Comment:
Seems we can't just ignore the `--zone` flag, otherwise it acts like the flags have been removed.
We should migrate the flag value to `--zones`, and make sure `--zone` and `--zones` can't co-exist.

Reply:
it's okay because karmada-agent has not such flag `--zone`. it should be completed in pr https://github.com/karmada-io/karmada/pull/3844.

Reply:
Thanks for the reminder. That's a dead code.